src/tools/fuzzing.h

```diff
@@ -934,8 +934,13 @@ class TranslateToFuzzReader {
     // Pick a chance to fuzz the contents of a function.
     const int RESOLUTION = 10;
     auto chance = upTo(RESOLUTION + 1);
-    for (auto& ref : wasm.functions) {
-      auto* func = ref.get();
+    // Do not iterate on wasm.functions itself, as we may add to it as we go
+    // through the functions - make() can add new functions to implement a
+    // RefFunc. Looping on an index avoids an iterator invalidation, and also
```

Comment:
This first sentence no longer applies, since we're now iterating on wasm.functions itself but just avoiding using iterators.

Reply:
Good point, that's not clear enough. I meant "not iterating on `wasm.functions`" as "not doing `for (x : wasm.functions)` directly, as opposed to an index. Pushed a better wording (I hope).